src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -107,8 +107,8 @@ public void discoverClusterNodesAndSlots(Jedis jedis) {
     w.lock();
```

Comment:
```suggestions
List<Object> slots = jedis.clusterSlots();
w.lock();
```

Reply:
> Since the function discoverClusterNodesandSlots is part of the API for the class JedisClusterInfoCache, I think it is a good idea to add a better safety guarantee.

@adiamzn Agree, I think we can move `jedis.clusterSlots()` to the top of `w.lock()`, just like the `discoverClusterSlots` function.

Reply:
agreed, made the change

Reply:
@adiamzn `jedis.clusterSlots()` to the `top` of `w.lock()`, the lock time can be shortened.

Reply:
true, fixing